extension/src/support.c

```diff
@@ -0,0 +1,64 @@
+/* This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE for a copy of the license
+ */
+
+#include <postgres.h>
+#include <fmgr.h>
+
+#include <nodes/pathnodes.h>
+#include <nodes/supportnodes.h>
+
+#ifdef PG_MODULE_MAGIC
+PG_MODULE_MAGIC;
+#endif
+
+PG_FUNCTION_INFO_V1(const_support);
+Datum
```

Comment:
```suggestion
/* This is a support function that optimizes calls to the supported function if it's called with constant arguments.
     Such calls are transformed into a subselect of the function call. This allows the planner to make this call
     an InitPlan which is evaluated once per query instead of multiple times (e.g. on every tuple when the function is 
     used in a WHERE clause). This should be used on any stable function that is often called with constant
     arguments. 
*/
Datum
```